File: westcoast/src/main/java/frc/commands/auto/LoadingBayAutonomous.java
package frc.commands.auto;

import edu.wpi.first.wpilibj.Encoder;
import edu.wpi.first.wpilibj.drive.DifferentialDrive;
import edu.wpi.first.wpilibj2.command.WaitCommand;
import frc.commands.ballhandling.*;
import frc.subsystems.BallHandlingSubsystem;
import frc.subsystems.vision.*;
import frc.subsystems.vision.targeting.*;
import frc.maps.*;
import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;

public class LoadingBayAutonomous extends SequentialCommandGroup {

  public LoadingBayAutonomous(
    RobotMap map,
    DifferentialDrive drive,
    Encoder leftEncoder,
    Encoder rightEncoder,
    double distance,
    double angle,
    VisionRunner vision,
    BallHandlingSubsystem ballHandler
  ) {

    super(
      new LeaveLine(map, drive, leftEncoder, rightEncoder, distance),
      new WaitCommand(1),
      new PerfectTurn(map, drive, leftEncoder, rightEncoder, angle),
      new WaitCommand(1),
      new AlignToTarget(map, drive, vision),
      parallel(new ShooterLogic(ballHandler), new StayOnTarget(map, drive, vision))
    );
  }
}

File: westcoast/src/main/kotlin/frc/commands/auto/LoadingBayAutonomous.kt
package frc.commands.auto

import edu.wpi.first.wpilibj.Encoder
import edu.wpi.first.wpilibj.drive.DifferentialDrive
import edu.wpi.first.wpilibj2.command.SequentialCommandGroup
import edu.wpi.first.wpilibj2.command.WaitCommand
import frc.commands.drive.LeaveLine
import frc.commands.drive.PerfectTurn
import frc.robot.maps.RobotMap
import frc.subsystems.MagazineSubsystem
import frc.vision.VisionRunner
import frc.commands.targeting.AlignToTarget

class LoadingBayAutonomous(
    map: RobotMap,
    drive: DifferentialDrive,
    leftEncoder: Encoder,
    rightEncoder: Encoder,
    distance: Double,
    angle: Double,
    vision: VisionRunner?,
    ballHandler: MagazineSubsystem?
) : SequentialCommandGroup(
    LeaveLine(map, drive, leftEncoder, rightEncoder, distance),
    WaitCommand(1.0),
    PerfectTurn(map, drive, leftEncoder, rightEncoder, angle),
    WaitCommand(1.0),
    AlignToTarget(map, drive, vision!!)
)